src/server-extension/query/event.ts

```diff
@@ -5,13 +5,15 @@ export const buyEvent = `SELECT
     LEFT JOIN nft_entity ne on ne.id = e.nft_id
     WHERE e.interaction = 'BUY' AND ne.collection_id = $1;`
 
-export const collectionEventHistory = `SELECT
-	ce.id as collection_id,
+export const collectionEventHistory = (idList: string, dateRange: string) => `SELECT
+	ce.id as id,
 	DATE(e.timestamp),
 	count(e)
 FROM nft_entity ne
 JOIN collection_entity ce on ce.id = ne.collection_id
 JOIN event e on e.nft_id = ne.id
-WHERE e.interaction = 'BUY' and ce.id = $1 and e.timestamp >= NOW() - INTERVAL '30 DAY'
+WHERE e.interaction = 'BUY'
+and ce.id in (${idList})
```

Comment:
Why not ?

```suggestion
and ce.id in $1
```

Reply:
It does not work, I tried. 

The only way to make it works is turning it into a brutual string, and has to use `'`

Reply:
OKI thanks for the info

Reply:
I think this is a bug on `typeorm`, it compiles graphql wrongly into raw SQL.

this is another same bug when I implement passion filter for landing page, just in case you would meet the same problem.

![Screen Shot 2022-05-16 at 1 19 15 PM](https://user-images.githubusercontent.com/11361147/168524820-5ab16d60-86ed-4430-a9c9-8355974a6bf3.png)

Reply:
but if the `issuer_in` becomes `[""]`, the query would run well.